src/pybind/mgr/dashboard_v2/tools.py

```diff
@@ -486,3 +491,57 @@ def run(self):
         self.notify_listeners(self._queue)
         self._queue.clear()
         logger.debug("notification queue finished")
+
+
+def is_valid_ipv6_address(addr):
+    try:
+        socket.inet_pton(socket.AF_INET6, addr)
+        return True
+    except socket.error:
+        return False
+
+
+def build_url(host, scheme=None, port=None):
+    """
+    Build a valid URL. IPv6 addresses specified in host will be enclosed in brackets
+    automatically.
+
+    >>> build_url('example.com', 'https', 443)
+    'https://example.com:443'
```

Comment:
Can you make sure, this is tested? (add this file to tox.ini)

Reply:
@sebastian-philipp :+1: This revealed that the current implementation behaves differently on Python3 compared to Python2.